Link error under RHEL7 with ATS 9.1.1 and jemalloc.

@randall -- We had an issue with the linker when building ATS 9.1.1 under RHEL7. I had no problems with my development box, which is Ubuntu 20.04 LTS, but our build environment is still RHEL7 with the newer compiler installed. The problem was fixed when I reverted #8285 in our internal branch. We had previously built 9.0.2 and 9.1.0 with no issues as well. Any ideas?

```
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: /root/trafficserver-9.1.1/src/tscore/.libs/libtscore.so: undefined reference to symbol '_ZdlPv'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: /opt/jemalloc521/lib/libjemalloc.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

--- a/build/jemalloc.m4
+++ b/build/jemalloc.m4
@@ -54,7 +54,8 @@ if test "$has_jemalloc" != "no"; then
   jemalloc_has_libs=0
   if test "$jemalloc_base_dir" != "/usr"; then
     TS_ADDTO(CPPFLAGS, [-I${jemalloc_include}])
-    TS_ADDTO(LDFLAGS, [-L${jemalloc_ldflags} -Wl,-rpath,${jemalloc_ldflags}])
+    TS_ADDTO(LDFLAGS, [-L${jemalloc_ldflags}])
+    TS_ADDTO(LDFLAGS, [-Wl,--add-needed -L${jemalloc_ldflags} -Wl,-rpath,${jemalloc_ldflags} -Wl,--no-as-needed])
     TS_ADDTO_RPATH(${jemalloc_ldflags})
   fi
   # On Darwin, jemalloc symbols are prefixed with je_. Search for that first, then fall back

